src/graphics/texture.js

```diff
@@ -589,28 +589,34 @@ pc.extend(pc, function () {
                             ! (source[i] instanceof HTMLVideoElement))) { // not video
 
                             invalid = true;
+                            break;
                         }
                     }
                 } else {
                     // first face is missing
                     invalid = true;
                 }
 
-                for (i = 0; i < 6; i++) {
-                    if (invalid || this._levels[0][i] !== source[i])
-                        this._levelsUpdated[0][i] = true;
+                if (!invalid) {
```

Comment:
So the logic is a bit different here, and you don't change _levelsUpdated, if the texture is invalid.
What if _levelsUpdated was false, then you uploaded an invalid texture, and it's still false, yet we uploaded the black 4x4 data?

Reply:
@guycalledfrank 
Thanks for your review.

If texture is invalid, update _levelsUpdated at end of this method(line 616 in original file).
So skip this line.